### ISSUE TITLE
editorconfig naming rule violations do not throw warnings

### ISSUE THREAD

**Issue by JustASquid** [dotnet/roslyn#64697]
**Version Used**: 
dotnet 6.0.401

**Steps to Reproduce**:

- Create new template console project
- `dotnet new editorconfig`
- Add the following line to `.editorconfig` under `[*.cs]`: `dotnet_analyzer_diagnostic.severity = warning` - this should make all style violations result in warnings.
- Edit `Program.cs` to be something like 
```cs
var local_with_underscores = "test";
// Suppress unused variable warning 
Console.WriteLine(local_with_underscores);
```
- run `dotnet format -v d --verify-no-changes`

**Expected Behavior**:
Warning due to the incorrect naming convention

**Actual Behavior**:
No warning
